File: rust/src/models/custom_tax_rate.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CustomTaxRate {
    pub name: String,

    pub rate: String,

    pub tax_code: String,
}

impl CustomTaxRate {
    pub fn new(name: String, rate: String, tax_code: String) -> Self {
        Self {
            name,
            rate,
            tax_code,
        }
    }
}

File: typescript/src/models/customTaxRate.ts
// this file is @generated

export interface CustomTaxRate {
  name: string;

  rate: string;

  taxCode: string;
}

export const CustomTaxRateSerializer = {
  _fromJsonObject(object: any): CustomTaxRate {
    return {
      name: object["name"],
      rate: object["rate"],
      taxCode: object["tax_code"],
    };
  },

  _toJsonObject(self: CustomTaxRate): any {
    return {
      name: self.name,
      rate: self.rate,
      tax_code: self.taxCode,
    };
  },
};
